packages/cli/generators/repository/index.js

```diff
@@ -0,0 +1,398 @@
+    });
+    return super.setOptions();
+  }
+
+  checkLoopBackProject() {
+    return super.checkLoopBackProject();
+  }
+
+  /**
+   * get the property name for the id field
+   * @param {string} modelName
+   */
+  async _getModelIdType(modelName) {
+    let fileContent = '';
+    let modelFile = path.join(
+      this.artifactInfo.modelDir,
+      modelName.toLowerCase() + '.model.ts',
+    );
+    try {
+      fileContent = this.fs.read(modelFile, {});
+    } catch (err) {
+      debug(`${ERROR_READING_FILE} ${modelFile}: ${err.message}`);
+      return this.exit(err);
+    }
+
+    return tsquery.getIdFromModel(fileContent);
+  }
+
+  /**
+   * helper method to inspect and validate a repository type
+   */
+  async _inferRepositoryType() {
+    if (!this.artifactInfo.dataSourceClassName) {
+      return;
+    }
+    let result = this._isConnectorOfType(
+      KEY_VALUE_CONNECTOR,
+      this.artifactInfo.dataSourceClassName,
+    );
+    debug(`KeyValue Connector: ${result}`);
+
+    if (result) {
+      this.artifactInfo.repositoryTypeClass = KEY_VALUE_REPOSITORY;
+      this.artifactInfo.defaultTemplate = REPOSITORY_KV_TEMPLATE;
+    } else {
+      this.artifactInfo.repositoryTypeClass = DEFAULT_CRUD_REPOSITORY;
+      this.artifactInfo.defaultTemplate = REPOSITORY_CRUD_TEMPLATE;
+    }
+
+    // assign the data source name to the information artifact
+    let dataSourceName = this.artifactInfo.dataSourceClassName
+      .replace('Datasource', '')
+      .toLowerCase();
+
+    Object.assign(this.artifactInfo, {
+      dataSourceName: dataSourceName,
+    });
+  }
+
+  /**
+   * load the connectors available and check if the basedModel matches any
+   * connectorType supplied for the given connector name
+   * @param {string} connectorType single or a comma separated string array
+   */
+  _isConnectorOfType(connectorType, dataSourceClassName) {
+    debug(`calling isConnectorType ${connectorType}`);
+    let jsonFileContent = '';
+    let result = false;
+
+    if (!dataSourceClassName) {
+      return false;
+    }
+    let datasourceJSONFile = path.join(
+      this.artifactInfo.datasourcesDir,
+      dataSourceClassName
+        .replace('Datasource', '.datasource.json')
+        .toLowerCase(),
+    );
+
+    try {
+      jsonFileContent = this.fs.readJSON(datasourceJSONFile, {});
+    } catch (err) {
+      debug(`${ERROR_READING_FILE} ${datasourceJSONFile}: ${err.message}`);
+      return this.exit(err);
+    }
+
+    for (let connector of Object.values(connectors)) {
+      const matchedConnector =
+        jsonFileContent.connector === connector.name ||
+        jsonFileContent.connector === `loopback-connector-${connector.name}`;
+
+      if (matchedConnector && connectorType.includes(connector.baseModel)) {
+        result = true;
+        break;
+      }
+    }
+
+    return result;
+  }
+
+  _setupGenerator() {
+    this.artifactInfo = {
+      type: 'repository',
+      rootDir: 'src',
+    };
+
+    this.artifactInfo = {
```

Comment:
IIUC, this line is overriding `this.artifactInfo` defined with `type: repository` above, and changes it to `type:datasource`. Is intentional?